protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,253 @@
+        (address bidToken,) = _getBidAndAuctionTokens();
+        address _recipient = auctionConfigs[epochId].recipient;
+        IERC20(bidToken).safeTransferFrom(msg.sender, _recipient, amount);
+        depositors[msg.sender][epochId] += amount;
+
+        EpochInfo storage info = epochs[epochId];
+        info.totalBidTokenAmount += amount;
+        emit Deposit(msg.sender, epochId, amount);
+    }
+
+    /**
+     * @notice Claim (retro) rewards for an epoch . Cannot claim for a live epoch auction
+     * @param epochId Epoch to claim for
+     */
+    function claim(uint256 epochId) external virtual override {
+        /// @notice cannot claim for current live epoch
+        EpochInfo memory info = epochs[epochId];
+        if (info.startTime == 0) { revert InvalidEpoch(); }
+        if (info.endTime >= block.timestamp) { revert CannotClaim(epochId); }
+
+        uint256 bidTokenAmount = depositors[msg.sender][epochId];
+        if (bidTokenAmount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        delete depositors[msg.sender][epochId];
+        SpiceAuctionConfig storage config = auctionConfigs[epochId];
+        (, address auctionToken) = _getBidAndAuctionTokens(config);
+
+        uint256 claimAmount = bidTokenAmount.mulDivRound(info.totalAuctionTokenAmount, info.totalBidTokenAmount, false);
+        /// checkpoint claim for auction token
+        _claimedAuctionTokens[auctionToken] += claimAmount;
+        IERC20(auctionToken).safeTransfer(msg.sender, claimAmount);
+        emit Claim(msg.sender, epochId, bidTokenAmount, claimAmount);
+    }
+
+    /**
+     * @notice Get spice auction config for an auction
+     * @param auctionId Id of auction
+     */
+    function getAuctionConfig(uint256 auctionId) external view override returns (SpiceAuctionConfig memory) {
+        return auctionConfigs[auctionId];
+    }
+
+    /**
+     * @notice Get auction token for current epoch
+     * @return Auction token
+     */
+    function getAuctionTokenForCurrentEpoch() external override view returns (address) {
+        SpiceAuctionConfig memory config = auctionConfigs[_currentEpochId];
+        return config.isTempleGoldAuctionToken ? templeGold : spiceToken;
+    }
+
+    /**
+     * @notice Get current epoch
+     * @return Epoch Id
+     */
+    function currentEpoch() external view override returns (uint256) {
+        return _currentEpochId;
+    }
+
+    /**
+     * @notice Get claimable amount for an epoch
+     * @dev For current epoch, function will return claimable at current time. This can change with more user deposits
+     * @param depositor Address to check amount for
+     * @param epochId Epoch id
+     * @return Claimable amount
+     */
+    function getClaimableAtCurrentTimestamp(address depositor, uint256 epochId) external override view returns (uint256) {
+        uint256 bidTokenAmount = depositors[depositor][epochId];
+        if (bidTokenAmount == 0 || epochId > _currentEpochId) { return 0; }
```

Comment:
Should this return zero if the auction for epochId is currently running? Or is the intent that this will continue to tick down as others deposit through the auction?

Reply:
yes, the claimable for the running epoch changes. so the intent is to not provide any false positive until epoch is ended